### PR TITLE
Auto-focus window renamer textbox on open

### LINKED ISSUE #11322: Rename window command should automatically focus the new name textbox
### Windows Terminal version (or Windows build number)

1.11.2421

### Other Software

_No response_

### Steps to reproduce

Run the `Rename window...` command from command palette, focus is lost from the window until something is clicked.

### Expected Behavior

The new name textbox should be focused.

### Actual Behavior

Nothing in the window is focused until I click into it with my mouse.

### PR DESCRIPTION
**Pull request by zadjii-msft**
Does what it says on the tin. This is maximal BODGE. 

`TeachingTip` doesn't provide an `Opened` event.
(https://github.com/microsoft/microsoft-ui-xaml/issues/1607). But we
want to focus the renamer text box when it's opened. We can't do that
immediately, the TextBox technically isn't in the visual tree yet. We
have to wait for it to get added some time after we call IsOpen. How do
we do that reliably? Usually, for this kind of thing, we'd just use a
one-off LayoutUpdated event, as a notification that the TextBox was
added to the tree. HOWEVER:
* The _first_ time this is fired, when the box is _first_ opened,
  yeeting focus doesn't work on the first LayoutUpdated. It does work on
  the second LayoutUpdated. Okay, so we'll wait for two LayoutUpdated
  events, and focus on the second.
* On subsequent opens: We only ever get a single LayoutUpdated. Period.
  But, you can successfully focus it on that LayoutUpdated.

So, we'll keep track of how many LayoutUpdated's we've _ever_ gotten. If
we've had at least 2, then we can focus the text box.

We're also not using a ContentDialog for this, because in Xaml Islands a
text box in a ContentDialog won't receive _any_ keypresses. Fun!

## References
* microsoft/microsoft-ui-xaml#1607
* microsoft/microsoft-ui-xaml#6910
* microsoft/microsoft-ui-xaml#3257
* microsoft/terminal#9662

## PR Checklist
* [x] Will close out #12021, but that's an a11y bug that needs secondary
  validation
* [x] Closes #11322
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

## Validation Steps Performed

Tested manually

### CODE REVIEW

File: src/cascadia/TerminalApp/AppActionHandlers.cpp [hunk 1]
```diff
@@ -844,18 +844,54 @@ namespace winrt::TerminalApp::implementation
...
+        //
+        // So, we'll keep track of how many LayoutUpdated's we've _ever_ gotten.
+        // If we've had at least 2, then we can focus the text box.
         //
         // We're also not using a ContentDialog for this, because in Xaml
         // Islands a text box in a ContentDialog won't receive _any_ keypresses.
         // Fun!
         // WindowRenamerTextBox().Focus(FocusState::Programmatic);
+        _renamerLayoutUpdatedRevoker.revoke();
+        _renamerLayoutUpdatedRevoker = WindowRenamerTextBox().LayoutUpdated(winrt::auto_revoke, [weakThis = get_weak()](auto&&, auto&&) {
```
Comment (DHowett) [inline src/cascadia/TerminalApp/AppActionHandlers.cpp:873]: i believe assigning over this thing _does_ call revoke

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalApp/AppActionHandlers.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppActionHandlers.cpp:873]: I figured, but who knows how often we're burned by that

**Review by carlos-zamora — APPROVED** [review]
It all looks good to me! Thanks for doing this. 🙂

So is the plan that once the `Opened()` event gets added, we'll replace the `LayoutUpdated` section in `AppActionHandlers.cpp`? Should we file a follow-up? Might be good to keep track of things we'd like to do after ingesting WinUI 2.8 or something like that.

**Comment by DHowett** [comment]
@msftbot merge this in 1 minute

**Comment by zadjii-msft** [comment]
> Should we file a follow-up

Well, so, **I** filed the PR to add `Opened` to MUX, so I REALLY doubt I'll forget to include that one 😜

**Pull request merged**